jboss-eap-config-elytron/added/launch/elytron.sh

```diff
@@ -0,0 +1,261 @@
+            </key-store>"
+    echo ${key_store}
+}
+
+create_elytron_keymanager() {
+    declare key_manager="$1" key_store="$2" key_password="$3"
+    # note key password here may be the same as the password to the keystore itself, or a seperate key specific password.
+    # in either case it is required.
+    local key_password="<credential-reference clear-text=\"${key_password}\"/>"
+    local elytron_keymanager="<key-manager name=\"${key_manager}\" key-store=\"${key_store}\">$key_password</key-manager>"
+    echo ${elytron_keymanager}
+}
+
+create_elytron_ssl_context() {
+    declare ssl_context_name="$1" key_manager_name="$2"
+    echo "<server-ssl-context name=\"${ssl_context_name}\" key-manager=\"${key_manager_name}\"/>"
+}
+
+create_elytron_https_connector() {
+    declare name="$1" socket_binding="$2" ssl_context="$3" proxy_address_forwarding="$4"
+    echo "<https-listener name=\"${name}\" socket-binding=\"${socket_binding}\" ssl-context=\"${ssl_context}\" proxy-address-forwarding=\"${proxy_address_forwarding:-true}\"/>"
+}
+
+configure_https() {
+
+  if [ "${CONFIGURE_ELYTRON_SSL}" != "true" ]; then
+    echo "Using PicketBox SSL configuration."
+    return 
+  fi
+
+  local ssl="<!-- No SSL configuration discovered -->"
+  local https_connector="<!-- No HTTPS configuration discovered -->"
+  local missing_msg="WARNING! Partial HTTPS configuration, the https connector WILL NOT be configured. Missing:"
+  local key_password=""
+  local elytron_key_store=""
+  local elytron_key_manager=""
+  local elytron_server_ssl_context=""
+  local elytron_https_connector=""
+
+  if [ -n "${HTTPS_PASSWORD}" -a -n "${HTTPS_KEYSTORE}" -a -n "${HTTPS_KEYSTORE_TYPE}" ]; then
+    if [ -n "${HTTPS_KEY_PASSWORD}" ]; then
+      key_password="${HTTPS_KEY_PASSWORD}"
+    else
+      echo "No HTTPS_KEY_PASSWORD was provided; using HTTPS_PASSWORD for Elytron LocalhostKeyManager."
+      key_password="${HTTPS_PASSWORD}"
+    fi
+
+    local elytron_key_store=$(create_elytron_keystore "LocalhostKeyStore" "${HTTPS_KEYSTORE}" "${HTTPS_PASSWORD}" "${HTTPS_KEYSTORE_TYPE}" "${HTTPS_KEYSTORE_DIR}")
+    local elytron_key_manager=$(create_elytron_keymanager "LocalhostKeyManager" "LocalhostKeyStore" "${key_password}")
+    local elytron_server_ssl_context=$(create_elytron_ssl_context "LocalhostSslContext" "LocalhostKeyManager")
+    local elytron_https_connector=$(create_elytron_https_connector "https" "https" "LocalhostSslContext" "true")
+  else
+    if [ -z "${HTTPS_PASSWORD}" ]; then
+      missing_msg="$missing_msg HTTPS_PASSWORD"
+    fi
+    if [ -z "${HTTPS_KEYSTORE}" ]; then
+      missing_msg="$missing_msg HTTPS_KEYSTORE"
+    fi
+    if [ -z "${HTTPS_KEYSTORE_TYPE}" ]; then
+      missing_msg="$missing_msg HTTPS_KEYSTORE_TYPE"
+    fi
+    echo ${missing_msg}
+  fi
+
+  # check for new config tag, use that if it's present
+  if [ "$(has_elytron_tls "${CONFIG_FILE}")" = "true" ]; then
+    # has new tag, replace it
+    insert_elytron_tls
+  fi
```

Comment:
I don't think the L215-219 work should happen outside the L190 'if'. The other sedding below just replaces ## comment elements with an empty string if the L190 case wasn't entered, so that's harmless. But this replaces a comment with a real but unusual, and thus prone to breakage, xml structure. (Unusual == a bunch of wrapper elements with nothing inside the wrapper.)

Reply:
Yeah the empty structure thing could be an issue. I did test it early on in EAP and it doesn't do anything unexpected when it's empty, but random breakage could result in the future.